java/src/jmri/jmrix/zimo/Mx1SystemConnectionMemo.java

```diff
@@ -83,11 +83,11 @@ public boolean provides(Class<?> type) {
         if (getDisabled()) {
             return false;
         }
-        if (type.equals(jmri.GlobalProgrammerManager.class)) {
-            return getProgrammerManager().isGlobalProgrammerAvailable();
+        if (type.equals(jmri.GlobalProgrammerManager.class)) {            
+            return true;
         }
-        if (type.equals(jmri.AddressedProgrammerManager.class)) {
-            return getProgrammerManager().isAddressedModePossible();
+        if (type.equals(jmri.AddressedProgrammerManager.class)) {            
+            return false;
```

Comment:
I found the above to lines were the only place that the ProgrammerManger was used before the traffic controller has existed,  Returning true or false directly here allows the creation of the program manager to be deferred and also seems consistent with what was done in this method for the other manager types.

Reply:
I don’t think the problem is that the traffic controller doesn’t exist, but that the programmer manager doesn’t exist the first time the provided method is called.

This fix is consistent with other systems that use this configuration approach however.